ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt

```diff
@@ -80,4 +80,39 @@ class SurveyRepositoryTest : BaseHiltTest() {
       surveyRepository.activeSurvey.test().assertValue(Optional.of(SURVEY))
       assertThat(fakeRemoteDataStore.isSubscribedToSurveyUpdates(SURVEY.id)).isFalse()
     }
+
+  @Test
+  fun deleteSurvey_whenSurveyIsActive() =
+    runTest(testDispatcher) {
+      surveyStore.insertOrUpdateSurvey(SURVEY).await()
+      surveyRepository.activateSurvey(SURVEY.id)
+      advanceUntilIdle()
+
+      surveyRepository.deleteSurvey(SURVEY.id)
+      advanceUntilIdle()
+
+      // Verify survey is deleted
+      surveyStore.surveys.test().assertValues(listOf())
```

Comment:
Can we use public methods of `SurveyRepository` here and below to verify rather than relying on the backing implementation? It could make the tests more readable and less brittle.

Reply:
@shobhitagarwal1612 Could you do this for the tests you added? No need to retroactively apply this comment.

Reply:
I applied the suggestion to this class. Did I miss any occurrence?

Reply:
Done